python/across_api/across/resolve.py

```diff
@@ -0,0 +1,135 @@
+    """
+    search_query = json.dumps(
+        {"query": {"bool": {"filter": {"term": {"properties.ztf_object_id": ztf_id}}}}}
+    )
+
+    params = {
+        "sort": "-properties.newest_alert_observation_time",
+        "elasticsearch_query[locus_listing]": search_query,
+    }
+    r = requests.get(ANTARES_URL, params=params)
+
+    if r.status_code == 200:
+        antares_data = json.loads(r.text)
+        ra = antares_data["data"][0]["attributes"]["ra"]
+        dec = antares_data["data"][0]["attributes"]["dec"]
+        return ra, dec
+    else:
+        return None, None
+
+
+class Resolve(ACROSSAPIBase):
+    """
+    Resolve class for resolving astronomical object names.
+
+    Parameters:
+    -----------
+    name : str
+        The name of the astronomical object to resolve.
+
+    Attributes:
+    -----------
+    ra : Optional[float]
+        The right ascension of the resolved object.
+    dec : Optional[float]
+        The declination of the resolved object.
+    name : str
+        The name of the astronomical object.
+    resolver : Optional[str]
+        The resolver used for resolving the object.
+
+    Methods:
+    --------
+    get() -> bool:
+        Retrieves the resolved object information.
+    """
+
+    api_name: str = "Resolve"
+    _schema = ResolveSchema
+    _get_schema = ResolveGetSchema
+
+    # Type hints
+    ra: Optional[float]
+    dec: Optional[float]
+    name: str
+    resolver: Optional[str]
+
+    def __init__(self, name: str):
+        # Class specific values
+        self.ra = None
+        self.dec = None
+        self.name = name
+        self.resolver = None
+        if self.validate_get():
+            self.get()
+
+    def get(self) -> bool:
+        """
+        Retrieves the RA and Dec coordinates for a given name.
+
+        Returns
+        -------
+        bool
+            True if the name is successfully resolved, False otherwise.
+
+        Raises
+        ------
+        HTTPException
+            If the name couldn't be resolved.
+        """
+        # Make sure the required parameters are given in the correct format
+        if not self.validate_get():
+            return False
+
+        """Do a name search"""
+        # Check against the ANTARES broker
+        if "ZTF" in self.name:
+            ra, dec = antares_radec(self.name)
+            if ra is not None:
+                self.ra, self.dec = ra, dec
+                self.resolver = "ANTARES"
+                return True
+
+        # Check using the CDS resolver
+        try:
+            skycoord = SkyCoord.from_name(self.name)
+            self.ra, self.dec = skycoord.ra.deg, skycoord.dec.deg
+            self.resolver = "CDS"
+            return True
+        except NameResolveError:
+            pass
+
+        # Send a warning if name couldn't be resolved
+        raise HTTPException(status_code=404, detail="Could not resolve name.")
```

Comment:
Hmm, 404 not found doesn't seem like the right status code here.

Reply:
Haha! I did agonize over this one. 404 is probably a stretch I'll admit, but it's not clear to which which status code to communicate that the request did not work because the name of the object you typed in was not recognized or "Not Found". Suggestions welcome.

Reply:
BTW it strikes me that 404 might be appropriate if we change the resolve endpoint to work like this:

`/across/resolve/Crab`

rather than:

`/across/resolve?name=Crab`

For the former I think a 404 is appropriate, and it's a small change.

Reply:
How about just returning a JSON response that omits the ra and dec properties?

Reply:
b927619 reverts the code to how I handle this in the Swift API, which does what you suggest.